### PR TITLE
Respect tab stop width and tab vs space in fakevim mode

### PR DESCRIPTION
**Pull request by Waqar144**
https://github.com/pbek/QOwnNotes/issues/1875

I have tried very hard to fix "Ctrl + V", it deactivates in fakeVim correctly but for some reason the text never gets pasted. It is probably because of some shortcut conflict or some other thing. Not sure. Any ideas?

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -602,6 +603,16 @@ void MainWindow::initFakeVim(QOwnNotesMarkdownTextEdit *noteTextEdit) {
 
     auto proxy = new FakeVimProxy(noteTextEdit, this, handler);
 
+    QSettings settings;
+    if (settings.value(QStringLiteral("Editor/useTabIndent")).toBool()) {
+        FakeVim::Internal::theFakeVimSettings()->item("et")->setValue(true);
```
Comment (pbek) [inline src/mainwindow.cpp:608]: Can't you use `FakeVim::Internal::theFakeVimSettings()->item("et")->setValue(VARIABLE_HERE);` to get around code duplication?

**Comment by pbek** [comment]
> I have tried very hard to fix "Ctrl + V", it deactivates in fakeVim correctly but for some reason the text never gets pasted. 

Maybe https://github.com/pbek/qmarkdowntextedit/blob/dbccd4a2afda9f12a463269a8c6768076834d254/qmarkdowntextedit.cpp#L252-L261 ?

File: src/mainwindow.cpp (hunk 1 above)
Comment (pbek) [reply src/mainwindow.cpp:608]: Wait, what are those two lines of `FakeVim::Internal::theFakeVimSettings()->item("et")->setValue(true);` doing? 😆

Comment (pbek) [reply src/mainwindow.cpp:608]: Both times true?

Comment (Waqar144) [reply src/mainwindow.cpp:608]: > Can't you use `FakeVim::Internal::theFakeVimSettings()->item("et")->setValue(VARIABLE_HERE);` to get around code duplication?

Yes, I guess

Comment (Waqar144) [reply src/mainwindow.cpp:608]: > Wait, what are those two lines of `FakeVim::Internal::theFakeVimSettings()->item("et")->setValue(true);` doing? laughing

AH yes, one needs to be false.

**Comment by Waqar144** [comment]
> > I have tried very hard to fix "Ctrl + V", it deactivates in fakeVim correctly but for some reason the text never gets pasted.
> 
> Maybe https://github.com/pbek/qmarkdowntextedit/blob/dbccd4a2afda9f12a463269a8c6768076834d254/qmarkdowntextedit.cpp#L252-L261 ?

Tested that already, the problem seems to be elsewhere :thinking:

**Pull request merged**